osu.Game.Rulesets.Catch/Skinning/Legacy/LegacyFruitPiece.cs

```diff
@@ -1,83 +1,39 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
-using osu.Framework.Allocation;
 using osu.Framework.Bindables;
-using osu.Framework.Graphics;
-using osu.Framework.Graphics.Containers;
-using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.Textures;
 using osu.Game.Rulesets.Catch.Objects.Drawables;
-using osu.Game.Rulesets.Catch.UI;
-using osu.Game.Rulesets.Objects.Drawables;
-using osu.Game.Skinning;
-using osuTK;
-using osuTK.Graphics;
 
 namespace osu.Game.Rulesets.Catch.Skinning.Legacy
 {
-    internal class LegacyFruitPiece : CompositeDrawable
+    internal class LegacyFruitPiece : LegacyCatchHitObjectPiece
     {
-        private readonly string lookupName;
+        public readonly Bindable<FruitVisualRepresentation> VisualRepresentation = new Bindable<FruitVisualRepresentation>();
 
-        private readonly Bindable<Color4> accentColour = new Bindable<Color4>();
-        private readonly Bindable<bool> hyperDash = new Bindable<bool>();
-        private Sprite colouredSprite;
-
-        public LegacyFruitPiece(string lookupName)
+        private readonly string[] lookupNames =
```

Comment:
```suggestion
        private static readonly string[] lookupNames =
```
However, is there a reason you did it this way instead of a switch?

Reply:
I'll make it a switch